mono/metadata/debug-mono-ppdb.c

```diff
@@ -595,3 +596,130 @@ mono_ppdb_lookup_locals (MonoDebugMethodInfo *minfo)
 
 	return res;
 }
+
+/*
+* We use this to pass context information to the row locator
+*/
+typedef struct {
+	int idx;			/* The index that we are trying to locate */
+	int col_idx;		/* The index in the row where idx may be stored */
+	MonoTableInfo *t;	/* pointer to the table */
+	guint32 result;
+} locator_t;
+
+static int
+table_locator (const void *a, const void *b)
+{
+	locator_t *loc = (locator_t *)a;
+	const char *bb = (const char *)b;
+	guint32 table_index = (bb - loc->t->base) / loc->t->row_size;
+	guint32 col;
+
+	col = mono_metadata_decode_row_col(loc->t, table_index, loc->col_idx);
+
+	if (loc->idx == col) {
+		loc->result = table_index;
+		return 0;
+	}
+	if (loc->idx < col)
+		return -1;
+	else
+		return 1;
+}
+
+typedef gboolean (*GuidComparer) (guint8* guid);
+static gboolean
+is_async_method_stepping_information_guid (guint8* debug_info_kind_guid)
+{
+	return debug_info_kind_guid [0] == 0xC5 &&
+		debug_info_kind_guid [1] == 0x2A &&
+		debug_info_kind_guid [2] == 0xFD &&
+		debug_info_kind_guid [3] == 0x54 &&
+		debug_info_kind_guid [4] == 0x25 &&
+		debug_info_kind_guid [5] == 0xE9 &&
+		debug_info_kind_guid [6] == 0x1A &&
+		debug_info_kind_guid [7] == 0x40 &&
+		debug_info_kind_guid [8] == 0x9C &&
+		debug_info_kind_guid [9] == 0x2A &&
+		debug_info_kind_guid [10] == 0xF9 &&
+		debug_info_kind_guid [11] == 0x4F &&
+		debug_info_kind_guid [12] == 0x17 &&
+		debug_info_kind_guid [13] == 0x10 &&
+		debug_info_kind_guid [14] == 0x72 &&
+		debug_info_kind_guid [15] == 0xF8;
+}
+
+//for parent_type see HasCustomDebugInformation table at https://github.com/dotnet/roslyn/blob/2ae8d5fed96ab3f1164031f9b4ac827f53289159/docs/specs/PortablePdb-Metadata.md#customdebuginformation-table-0x37
```

Comment:
the url is this:
https://github.com/dotnet/corefx/blob/master/src/System.Reflection.Metadata/specs/PortablePdb-Metadata.md